### PR TITLE
Ofir/rebase on upstream

### PR DESCRIPTION
**Pull request by ofir-amir**
1. Allow PBR versioning scheme when pushing to registry (So we can have 1.2.3.<shorthash> or 4.5.6.patch.<shorthash> as a version
2. Support authenticated Registry
3. Add the hash as tag argument so use can build an image which is based on another image on the same build (So we can have base image and multiple derived images with the same tag) 
4. Better support for Mac (Not finalized yet - some issue the docker user which does not exists on mac - maybe fixed by what you did for podman - need to check)

### CODE REVIEW

**Review by eranco74 — APPROVED** [review]
LGTM

File: skipper/runner.py [hunk 1]
```diff
@@ -104,10 +111,12 @@ def handle_volumes_bind_mount(docker_cmd, homedir, volumes, workspace):
     else:
         volumes.extend([
             '%(workspace)s:%(workspace)s:rw,Z' % dict(workspace=workspace),
-            '/var/lib/osmosis:/var/lib/osmosis:rw,Z',
             '/var/run/docker.sock:/var/run/docker.sock:Z',
             '%s:/opt/skipper/skipper-entrypoint.sh:Z' % utils.get_extra_file("skipper-entrypoint.sh"),
             ])
+        # Will fail on Mac
+        if os.path.exists('/var/lib/osmosis'):
```
Comment (tsorya) [inline skipper/runner.py:117]: lets move it from if else, looks like it is the same in podman and docker

File: skipper/runner.py [hunk 2]
```diff
@@ -92,6 +93,12 @@ def handle_volumes_bind_mount(docker_cmd, homedir, volumes, workspace):
     volumes.extend(['%(homedir)s/.netrc:%(homedir)s/.netrc:ro' % dict(homedir=homedir),
                     '%(homedir)s/.gitconfig:%(homedir)s/.gitconfig:ro' % dict(homedir=homedir)])
 
+    # required for docker login
+    if os.path.exists('%(homedir)s/.docker/config.json' % dict(homedir=homedir)):
```
Comment (tsorya) [inline skipper/runner.py]: https://github.com/Stratoscale/skipper/blob/upstream/skipper/runner.py#L140
there is special treatment for this mount. Maybe let's always add it?

File: skipper/runner.py (hunk 1 above)
Comment (ofir-amir) [reply skipper/runner.py:117]: @tsorya but it is different with podman we don't use :Z
Don't know enough about podman but with docker two concurrent skipper run will fail

File: skipper/runner.py (hunk 2 above)
Comment (ofir-amir) [reply skipper/runner.py]: :+1:

**Review by tsorya — APPROVED** [review]

**Pull request merged**